index.html

```diff
@@ -293,80 +302,160 @@ <h4>The <dfn>ThingFilter</dfn> dictionary</h4>
-            <li>"<dfn>local</dfn>" for discovering <a>Thing</a>s defined in the same device</li>
-            <li>
-              "<dfn>nearby</dfn>" for discovering <a>Thing</a>s nearby the device, e.g. by Bluetooth or NFC
-            </li>
-            <li>
-              "<dfn>directory</dfn>" for discovery based on a service provided by a directory or repository of <a>Thing</a>s
-            </li>
-            <li>
-              "<dfn>broadcast</dfn>" for an open ended discovery based on sending a request to a broadcast address
-            </li>
-            <li>
-              "<dfn>other</dfn>" for a proprietary method defined by the solution.
-            </li>
-          </ul>
-        </p>
-      </section>
     </section> <!-- Discovery -->
 
+    <section> <h3> The <dfn>fetch()</dfn> method</h3>
+      <p>
+        Accepts an <code>url</code> argument and returns a <code><a>Promise</a></code> that resolves with a <a>ThingDescription</a>.
+      </p>
+    </section>
+
+    <section data-dfn-for="ThingDescription">
+    <h3>The <dfn>ThingDescription</dfn> type</h3>
+      <p>
+        Representation of the <a>Thing Description</a>, standardized in the [Wot Things Description](https://w3c.github.io/wot-thing-description/) specification.
+      </p>
+      <p class="note">
+        In this version of the API, <a>Thing Description</a>s are represented as opaque strings, denoting a serialized form, for instance JSON or JSON-LD. See [Issue 38](https://github.com/w3c/wot-scripting-api/issues/38) and [Issue 45](https://github.com/w3c/wot-scripting-api/issues/45).
+      </p>
+    </section>
+
     <section> <h3> The <dfn>consume()</dfn> method</h3>
       <p>
-        Accepts an <code>url</code> argument and returns a <code><a>Promise</a></code> of a <a>ConsumedThing</a>.
+        Accepts an <code>td</code> argument of type <code><a>ThingDescription</a></code> and returns a <a>ConsumedThing</a> object instantiated based on that description.
       </p>
     </section>
 
-    <section> <h3>The <dfn>expose()</dfn> method</h3>
+    <section> <h3>The <dfn id="produce-method">produce()</dfn> method</h3>
       <p>
-        Returns a <code><a>Promise</a></code> of a locally created <a>ExposedThing</a> based on the provided initialization parameters.
+        Accepts a <code>model</code> argument of type <code><a>ThingModel</a></code> and returns an <a>ExposedThing</a> object, locally created based on the provided initialization parameters. An <a>ExposedThing</a> can be created in the following ways:
+        <ul>
+          <li>
+            from an initial model (name and semantic annotations), then adding properties, actions, events and request handlers;
+          </li>
+          <li>
+            from a <a>Thing Description</a>, then adding request handlers;
+          </li>
+          <li>
+            from an existing <a>ConsumedThing</a> object, based on its <a>Thing Description</a>, then adding request handlers.
```

Comment:
The third option is a special case of option 2. Hence I propose to remove option 3 and change option 2 to 

from a Thing Description (possibly also retrieved from a ConsumedThing), then adding request handlers;

Reply:
That should be fine. I just wanted to highlight it is possible to expose a ConsumedThing.

Reply:
Done (not committed yet).